### PR TITLE
docs(configuration): add the explanation of Architecture.diff_* functions

### PR DESCRIPTION
**Pull request by taro-yu**
## Description
`Require differences of two architectures` is added to chapter Configuration of CARET_doc.
`Require differences of two architectures` section explains how to use `diff_XXX_YYY` function.

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

**Review by rokamu623 — CHANGES_REQUESTED** [review]
As pointed out in the github action, it is `Architecture`, not `Architecure`.

File: docs/configuration/architecture_diff_func.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+print(left_only_node_names)
+print(right_only_node_names)
+
+#result
+#('node2',)
+#('node3', 'node4')
+
+```
+
+In this setup, the `left_arch` has two nodes: `node1` and `node2`, and the `right_arch` has three nodes: `node1`, `node3` and `node4`. Therefore, the node name that `left_arch` and `right_arch` have in common is `node1`.
```
Comment (keita1523) [inline docs/configuration/architecture_diff_func.md:46]:
```suggestion
In this setup, the `left_arch` has two nodes: `node1` and `node2`, and the `right_arch` has three nodes: `node1`, `node3` and `node4`. 
The `left_arch` and `right_arch` have `node1` as common node.
```

File: docs/configuration/architecture_diff_func.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+```
+
+In this setup, the `left_arch` has two topic names: `topic1` and `topic2`, and the `right_arch` has three topic names: `topic1`, `topic3` and `topic4`. The topic name here is the name of the topic used for publish or subscribe.　Therefore, the topic name that `left_arch` and `right_arch` have in common is `topic1`.
+Therefore, the topic name that only `left_arch` has is `topic2`, and the topic names that only `right_arch` has are `topic3` and `topic4`, which are the return value and output at the end of the script respectively.
+
+## How to compare node in architecture
+
+<!-- This section describes the functions `diff_node_pubs()` and `diff_node_subs()`, which compare the contents of two nodes in architecture. -->
+
+`diff_node_pubs()` and `diff_node_subs()` are functions that take nodes in an architecture and compare them. The nodes given can be two nodes in the same architecture or in two different architectures. These functions compare the two given nodes and find the pub/sub topic name that exists only in each of them. Thus, what one node has but the other does not (pub/sub topic name) can be identified and compared.
```
Comment (keita1523) [inline docs/configuration/architecture_diff_func.md:75]:
```suggestion
Nodes consist of the node name, publish topic, and subscription topic.
The `diff_node_pubs()` and `diff_node_subs()` function find the publish and subscription topics that exist only in one of the two nodes. 

```

**Review by keita1523 — CHANGES_REQUESTED** [review]

File: docs/configuration/architecture_diff_func.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+#('node3', 'node4')
+
+```
+
+In this setup, the `left_arch` has two nodes: `node1` and `node2`, and the `right_arch` has three nodes: `node1`, `node3` and `node4`. Therefore, the node name that `left_arch` and `right_arch` have in common is `node1`.
+Therefore, the node name that only `left_arch` has is `node2`, and the node names that only `right_arch` has are `node3` and `node4`, which are the return value and output at the end of the script respectively.
+
+### diff_topic_names()
+
+An example of the use of `diff_topic_names()` is shown in `#sample_2`.
```
Comment (keita1523) [inline docs/configuration/architecture_diff_func.md:51]:
```suggestion
The function `diff_topics_names()` compares topic names of nodes.
```

File: docs/configuration/architecture_diff_func.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+#get topic names that are only in old and new respectively.
+left_only_topics, right_only_topics = Architecture.diff_topic_names(left_arch, right_arch)
+
+print(left_only_topics)
+print(right_only_topics)
+
+#result
+#('topic2',)
+#('topic3', 'topic4')
```
Comment (keita1523) [inline docs/configuration/architecture_diff_func.md:64]:
```suggestion
# get difference topic names
left_only_topics, right_only_topics = Architecture.diff_topic_names(left_arch, right_arch)

# display differences
print(left_only_topics)
print(right_only_topics)
```

File: docs/configuration/architecture_diff_func.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+```
+
+In this setup, the `left_arch` has two topic names: `topic1` and `topic2`, and the `right_arch` has three topic names: `topic1`, `topic3` and `topic4`. The topic name here is the name of the topic used for publish or subscribe.　Therefore, the topic name that `left_arch` and `right_arch` have in common is `topic1`.
+Therefore, the topic name that only `left_arch` has is `topic2`, and the topic names that only `right_arch` has are `topic3` and `topic4`, which are the return value and output at the end of the script respectively.
```
Comment (keita1523) [inline docs/configuration/architecture_diff_func.md:69]:
```suggestion
This function inputs two architectures, compares them, and outputs the topic names that exist only in one of nodes.
```

**Review by keita1523 — CHANGES_REQUESTED** [review]
Please apply this modification to other functions.

File: docs/configuration/architecture_diff_func.md (hunk 2 above)
Comment (keita1523) [reply docs/configuration/architecture_diff_func.md:75]: Don't need to apply this modification.
Please apply [here](https://github.com/tier4/CARET_doc/pull/168#pullrequestreview-1347730340).

File: docs/configuration/architecture_diff_func.md (hunk 1 above)
Comment (keita1523) [reply docs/configuration/architecture_diff_func.md:46]: Don't need to apply this modification.
Please apply [here](https://github.com/tier4/CARET_doc/pull/168#pullrequestreview-1347730340).

File: docs/configuration/.pages [hunk 6]
```diff
@@ -7,3 +7,4 @@ nav:
     - Rename sub-objects: rename_function.md
     - Practical example: practical_example.md
     - Visualize application structure: visualize_application_structure.md
+    - Differences of architectures: architecture_diff_func.md
```
[the remark this answers is not in the document]
Comment (taro-yu) [reply docs/configuration/.pages]: Changed as above.

File: docs/configuration/index.md [hunk 7]
```diff
@@ -23,5 +23,6 @@ In detail, the following sections are listed as below.
 - [**How to define intra-node data path**](./intra_node_data_path.md) will let you know what `message_context` is. This is advanced topic than others
 - [**How to rename sub-objects**](./rename_function.md) section will tell you how to use `rename_XXX()`
 - [**Practical example with CARET_demos**](./practical_example.md) will demonstrates configuration process on [`CARET_demos`](https://github.com/tier4/CARET_demos)
+- [**Require differences of two architectures**](./architecture_diff_func.md) will tell you how to use `diff_XXX_YYY()` functions
```
[the remark this answers is not in the document]
Comment (taro-yu) [reply docs/configuration/index.md]: Changed as above.

**Review by keita1523 — CHANGES_REQUESTED** [review]

File: docs/configuration/architecture_diff_func.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
+# How to get differences of architectures
+
+The diff function can be used to find the difference between two architectures. Specifically, the diff functions can find data that only exist in each, e.g. node name, pub/sub topic.
```
Comment (keita1523) [inline docs/configuration/architecture_diff_func.md:3]:
```suggestion
The diff function can be used to find the difference between two architectures. Specifically, the diff functions can find data that exist only in one of the architectures, such as node names, publish/subscribe topics, etc. 
```

File: docs/configuration/architecture_diff_func.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+print(left_only_topics)
+print(right_only_topics)
+
+```
+
+This function inputs two architectures, compares them, and outputs the pub/sub topic names that exist only in one of architectures.
+
+<prettier-ignore-start>
+!!!info
+      The architecture object has other elements such as Executor and Callback. However, the functions that gets the difference of Executors or Callbacks are not yet implemented. Please contact the developers if needed.
```
Comment (keita1523) [inline docs/configuration/architecture_diff_func.md:58]:
```suggestion
      The architecture object has other elements such as Executor and Callback. However, the functions that get the difference of Executors or Callbacks are not yet implemented. Please contact the developers if needed.
```

**Review by keita1523 — DISMISSED** [review]
LGTM!
(Please pass pre-commit error)

**Review by rokamu623 — APPROVED** [review]
Sorry for the delay in approval.
LGTM!

**Pull request merged**